src/overlays/actors/ovl_En_Po_Sisters/z_en_po_sisters.c

```diff
@@ -21,127 +72,1329 @@ const ActorInit En_Po_Sisters_InitVars = {
+        if (this->unk_195 == 0) {
+            func_800F5ACC(0x38);
+        }
+        func_80AD9F1C(this);
+    } else {
+        this->actor.posRot.pos.y += 0.1f;
+        temp = this->unk_195;
+        if (temp != 0) {
+            if (this->unk_19A > 90) {
+                phi_v0 = 1;
+                phi_a2 = 64;
+            } else if (this->unk_19A > 70) {
+                phi_v0 = 0;
+                phi_a2 = 64;
+            } else if (this->unk_19A > 55) {
+                phi_v0 = 1;
+                phi_a2 = 96;
+            } else if (this->unk_19A > 40) {
+                phi_v0 = 0;
+                phi_a2 = 96;
+            } else {
+                phi_v0 = 1;
+                phi_a2 = 256;
+            }
+            if (this->unk_195 == 2) {
+                phi_a2 *= 2;
+            }
+            Math_ApproxUpdateScaledS(&this->actor.shape.rot.y,
+                                     this->actor.parent->shape.rot.y + (this->unk_195 * 0x4000) * phi_v0, phi_a2);
+        } else if (this->unk_19A == 70 || this->unk_19A == 40) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_LAUGH2);
+        }
+    }
+    func_80AD97C8(this, globalCtx);
+    Actor_SetHeight(&this->actor, 40.0f);
+}
+
+void func_80ADB770(EnPoSisters* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
+    s32 phi_a0;
+
+    if (this->unk_19A != 0) {
+        this->unk_19A--;
+    }
+    if (this->unk_19C > 0) {
+        if (this->unk_19A >= 16) {
+            SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+            if (this->unk_195 == 0) {
+                if (ABS((s16)(16 - this->unk_196)) < 14) {
+                    this->actor.shape.rot.y +=
+                        (0x580 - (this->unk_19C * 0x180)) * fabsf(Math_Sins(this->unk_196 * 0x800));
+                }
+                if (this->unk_19A >= 284 || this->unk_19A < 31) {
+                    this->unk_199 |= 0x40;
+                } else {
+                    this->unk_199 &= ~0x40;
+                }
+            } else {
+                this->actor.shape.rot.y = (s16)(this->actor.parent->shape.rot.y + (this->unk_195 * 0x4000));
+            }
+        }
+    }
+    if (this->unk_195 == 0) {
+        if (this->unk_19A >= 284 || (this->unk_19A < 31 && this->unk_19A >= 16)) {
+            this->unk_199 |= 0x40;
+        } else {
+            this->unk_199 &= ~0x40;
+        }
+    }
+    if (func_8002DBB0(&PLAYER->actor, &this->actor.initPosRot.pos) > 600.0f) {
+        this->unk_199 &= ~0x40;
+        func_80AD9C24(this, globalCtx);
+    } else if (this->unk_19A == 0) {
+        if (this->unk_195 == 0) {
+            func_80AD94E0(this);
+        } else {
+            func_80AD9C24(this, globalCtx);
+        }
+    } else if (this->unk_195 != 0) {
+        EnPoSisters* attached = (EnPoSisters*)this->actor.parent;
+        if (attached->actionFunc == func_80ADAAA4) {
+            func_80AD95D8(this);
```

Comment:
can prob rename now that its not attached

Reply:
ideally whatever it actually is, not just parent. but if you dont know thats fine